SteamKit2/SteamKit2/Steam/Handlers/SteamMatchmaking/Lobby.cs

```diff
@@ -0,0 +1,402 @@
+                public override bool Equals( object obj )
+                {
+                    if ( obj is Member member )
+                    {
+                        return SteamID.Equals( member.SteamID );
+                    }
+
+                    return false;
+                }
+
+                /// <summary>
+                /// Hash code of the lobby member. Only the SteamID of the lobby member is taken into account.
+                /// </summary>
+                /// <returns>The hash code of this lobby member.</returns>
+                public override int GetHashCode()
+                {
+                    return SteamID.GetHashCode();
+                }
+            }
+
+            /// <summary>
+            /// SteamID of the lobby.
+            /// </summary>
+            public SteamID SteamID { get; }
+
+            /// <summary>
+            /// The type of the lobby.
+            /// </summary>
+            public ELobbyType LobbyType { get; }
+
+            /// <summary>
+            /// The lobby's flags.
+            /// </summary>
+            public int LobbyFlags { get; }
+
+            /// <summary>
+            /// The SteamID of the lobby's owner. Please keep in mind that Steam does not provide lobby
+            /// owner details for lobbies returned in a lobby list. As such, lobbies that have been
+            /// obtained/updated as a result of calling <see cref="SteamMatchmaking.GetLobbyList"/>
+            /// may have a null (or non-null but state) owner.
+            /// </summary>
+            public SteamID OwnerSteamID { get; }
+
+            /// <summary>
+            /// The metadata of the lobby; string key-value pairs.
+            /// </summary>
+            public IReadOnlyDictionary<string, string> Metadata { get; }
+
+            /// <summary>
+            /// The maximum number of members that can occupy the lobby.
+            /// </summary>
+            public int MaxMembers { get; }
+
+            /// <summary>
+            /// The number of members that are currently occupying the lobby.
+            /// </summary>
+            public int NumMembers { get; }
+
+            /// <summary>
+            /// A list of lobby members. This will only be populated for the user's current lobby.
+            /// </summary>
+            public IReadOnlyList<Member> Members { get; }
+
+            /// <summary>
+            /// The distance of the lobby.
+            /// </summary>
+            public float? Distance { get; }
+
+            /// <summary>
+            /// The weight of the lobby.
+            /// </summary>
+            public long? Weight { get; }
+
+            static readonly IReadOnlyDictionary<string, string> EmptyMetadata =
+                new ReadOnlyDictionary<string, string>( new Dictionary<string, string>() );
+
+            static readonly IReadOnlyList<Member> EmptyMembers = new List<Member>().AsReadOnly();
```

Comment:
Nit: I'd prefer using `Array.Empty<Member>()` instead of a new list, but the difference is negligible.

Reply:
It'd have to be:

```C#
Array.AsReadOnly(Array.Empty<Member>());
```

However, yeah, I might as well change it seems as it's conceivable that user code may also use `Array.Empty<Member>()` and we'd get the benefit of just the one copy.

Reply:
Also, allocating an empty list gets you both a List<> object and an empty array underneath (though it's its own version instead of Array.Empty's). Using the empty array directly means one less object - the List.